dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -10,10 +10,18 @@
        for faster overall incremental processing.
   #} */
 
-  {% if language == 'sql' and (strategy in ('default', 'append', 'merge') or (unique_key is none)) %}
-    {{ return('view') }}
-  {% else %}  {#--  play it safe -- #}
-    {{ return('table') }}
+  {% if language != "sql" %}
+    {{ return("table") }}
+  {% elif strategy in ('default', 'merge') and merge_tmp_relation_type == "table"%}
```

Comment:
Jinja formatting nit to add missing whitespace:
```suggestion
  {% elif strategy in ('default', 'merge') and merge_tmp_relation_type == "table" %}
```